managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -3855,7 +3854,8 @@ static long estimateEntryCountBySize(long bytesSize, Position readPosition, Mana
                 posToRead = ml.getNextValidPosition(PositionFactory.create(posToRead.getLedgerId(), Long.MAX_VALUE));
             }
         }
-        return Math.max(result, 1);
+        int resultInt = Long.valueOf(result).intValue();
+        return Math.max(resultInt < 0 ? Integer.MAX_VALUE : resultInt, 1);
```

Comment:
A better solution would be to check the long value before converting it, since in theory, it could "wrap around" multiple times.
```suggestion
        int resultInt = result > Integer.MAX_VALUE ? Integer.MAX_VALUE : result;
        return Math.max(resultInt, 1);
```

Reply:
> A better solution would be to check the long value before converting it, since in theory, it could "wrap around" multiple times.

I changed the method return type from long to int, because we still need to convert it from long to int when using it.

Reply:
fixed